### PR TITLE
Member directory v1

### PR DESCRIPTION
**Pull request by Devorein**

### CODE REVIEW

File: hooks/useMemberProperties.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+};
+
+const MemberPropertiesContext = createContext<Readonly<Context>>({
+  properties: undefined,
+  addProperty: () => Promise.resolve({} as any),
+  updateProperty: () => Promise.resolve({} as any),
+  deleteProperty: () => Promise.resolve()
+});
+
+export function MemberPropertiesProvider ({ children }: { children: ReactNode }) {
```
Comment (mattcasey) [inline hooks/useMemberProperties.tsx:25]: double check we need this globally

Comment (msadura) [reply hooks/useMemberProperties.tsx:25]: Moved to member directory page level, good call

**Comment by Devorein** [comment]
1. I noticed that we have the `Status` column in the bounties nexus table, which seems more appropriate to be the action. It always shows `Review` as the action. Bounty have their own `status`, which is shown on the bounty's page
image
image
image

File: components/members/components/MemberDirectoryTableView.tsx [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+                      return (
+                        <TableCell>
+                          <Typography variant='body2'>{member.properties.find(_property => _property.memberPropertyId === timezoneProperty?.id)?.value ?? 'N/A'}</Typography>
+                        </TableCell>
+                      );
+                    }
+                    case 'name': {
+                      return (
+                        <TableCell>
+                          <Link href={`/u/${member.path}`}>
```
Comment (mattcasey) [inline components/members/components/MemberDirectoryTableView.tsx:97]: member.path is optional, should be `member.path || member.id`

File: components/members/MemberDirectoryPage.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+export default function MemberDirectoryPage () {
+  const { members } = useMembers();
+  const [searchedMembers, setSearchedMembers] = useState<Member[]>(members);
+  const { properties } = useMemberProperties();
+  const [currentView, setCurrentView] = useState<typeof views[number]>('gallery');
+  const [isPropertiesDrawerVisible, setIsPropertiesDrawerVisible] = useState(false);
+
+  return properties && searchedMembers ? (
+    <CenteredPageContent style={{
```
Comment (mattcasey) [inline components/members/MemberDirectoryPage.tsx:37]: this component has a min-height of 80vh and it makes the page look huge. I'm not sure this page needs a min-height, or why there is one on CenteredPageContent :P it also seems unecessary on the proposals page.

image

File: components/members/MemberDirectoryPage.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+const StyledButton = styled(Button)`
+  padding: ${({ theme }) => theme.spacing(0.5, 1)};
+
+  .Icon {
+    width: 20px;
+    height: 20px;
+  }
+`;
+
+const views = ['gallery', 'table'] as const;
```
Comment (mattcasey) [inline components/members/MemberDirectoryPage.tsx:27]: For "post v1" we should use the URL To track which view you're on. view=gallery|table, and gallery could just be the default.

File: components/members/components/AddMemberPropertyButton.tsx [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+  const { mutateMembers } = useMembers();
+  const propertyNamePopupState = usePopupState({ variant: 'popover', popupId: 'property-name-modal' });
+  const [selectedPropertyType, setSelectedPropertyType] = useState<null | MemberPropertyType>(null);
+  const [propertyName, setPropertyName] = useState('');
+  const { properties, addProperty } = useMemberProperties();
+  const [propertyOptions, setPropertyOptions] = useState<PropertyOption[]>([]);
+
+  return (
+    <>
+      <Button
```
Comment (mattcasey) [inline components/members/components/AddMemberPropertyButton.tsx:29]: Add Property should be aligned left (notion example below)

image

File: components/members/components/MemberDirectoryGalleryView.tsx [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+}: {
+  member: Member;
+}) {
+  const { properties = [] } = useMemberProperties();
+  const timezoneProperty = properties.find(property => property.type === 'timezone');
+  const nameProperty = properties.find(property => property.type === 'name');
+
+  return (
+    <Link
+      href={`/u/${member.path}`}
```
Comment (mattcasey) [inline components/members/components/MemberDirectoryGalleryView.tsx:24]: use member.id as a fallback for memberpath

File: components/members/components/MemberDirectoryTableView.tsx (hunk 2 above)
Comment (mattcasey) [reply components/members/components/MemberDirectoryTableView.tsx:97]: can we also add 'color=inherit'? the blue link color is unecessary

**Review by mattcasey — APPROVED** [review]
Approved but please fix the paths for members so that we use member.id as a fallback

File: components/members/components/MemberDirectoryTableView.tsx (hunk 2 above)
Comment (Devorein) [reply components/members/components/MemberDirectoryTableView.tsx:97]: Alright updating.

File: components/members/MemberDirectoryPage.tsx (hunk 4 above)
Comment (Devorein) [reply components/members/MemberDirectoryPage.tsx:27]: Updated. Just required a few lines

File: components/members/components/AddMemberPropertyButton.tsx (hunk 5 above)
Comment (Devorein) [reply components/members/components/AddMemberPropertyButton.tsx:29]: Updated. Thanks

File: components/members/MemberDirectoryPage.tsx (hunk 3 above)
Comment (Devorein) [reply components/members/MemberDirectoryPage.tsx]: This is a bit out of scope as the layout needs to change. A new ticket was created just for this

Comment (msadura) [reply components/members/MemberDirectoryPage.tsx]: https://app.charmverse.io/charmverse/page-7098501094691398 - ticket with description for the issue

**Pull request merged**